gcp_variant_transforms/libs/processed_variant.py

```diff
@@ -0,0 +1,283 @@
+objects should be used as non-mutable in other scopes, hence all mutating
+functions are "private".
+"""
+
+import logging
+from typing import Dict, List
+
+from gcp_variant_transforms.beam_io import vcfio
+from gcp_variant_transforms.libs import vcf_header_parser
+
+
+_FIELD_COUNT_ALTERNATE_ALLELE = 'A'
+
+
+class ProcessedVariant(object):
+  """A wrapper around the ``Variant`` class with extra functionality.
+
+  Given header file information, this can parse INFO fields that need to be
+  split and attached to alternates. This is not inherited from
+  :class:``vcfio.Variant`` as an encapsulation layer and to prefer composition.
+  """
+
+  def __init__(self, variant):
+    # type: (vcfio.Variant) -> None
+    if not isinstance(variant, vcfio.Variant):
+      raise ValueError('Expected an instance of vcfio.Variant.')
+    self._variant = variant
+    self._non_alt_infos = None
+    self._alternate_datas = [] # type: List[AlternateBaseData]
+    for a in variant.alternate_bases:
+      self._alternate_datas.append(AlternateBaseData(a))
+
+  def __repr__(self):
+    # TODO(bashir2): Add other processed fields too.
+    return self._variant.__repr__()
+
+  @property
+  def reference_name(self):
+    # type: () -> str
+    return self._variant.reference_name
+
+  @property
+  def start(self):
+    # type: () -> int
+    return self._variant.start
+
+  @property
+  def end(self):
+    # type: () -> int
+    return self._variant.end
+
+  @property
+  def reference_bases(self):
+    # type: () -> str
+    return self._variant.reference_bases
+
+  @property
+  def names(self):
+    # type: () -> List[str]
+    return self._variant.names
+
+  @property
+  def quality(self):
+    # type: () -> float
+    return self._variant.quality
+
+  @property
+  def filters(self):
+    # type: () -> List[str]
+    return self._variant.filters
+
+  @property
+  def calls(self):
+    # type: () -> List[vcfio.VariantCall]
+    return self._variant.calls
+
+  @property
+  def non_alt_infos(self):
+    # type: () -> Dict
+    return self._non_alt_infos
+
+  @property
+  def alternate_data_list(self):
+    # type: () -> List[AlternateBaseData]
+    if self._alternate_datas is None:
+      raise ValueError(
+          'This class should be constructed using the factory function.')
+    return self._alternate_datas
+
+
+class AlternateBaseData(object):
+  """This is to keep all information for a single alternate-bases."""
+
+  def __init__(self, alt_bases):
+    # type: (str) -> None
+    """
+    Args:
+      alt_bases(str): The alternate bases string for this instance.
+    """
+    self._alt_bases = alt_bases
+    # Note that `_infos` also holds the split annotation fields. For those
+    # fields, the value in the `_infos` dict has a list of dicts itself (in
+    # `vcfio.VariantInfo.data`).
+    self._infos = {}
+
+  @property
+  def alternate_bases(self):
+    # type: () -> str
+    return self._alt_bases
+
+  @property
+  def infos(self):
```

Comment:
I actually used 'info' in both Variant and VariantCall to be consistent with the proto definitions. I realize that this is more about proto semantics (which is why I was ok with using infos in VcfHeader), but you may consider being consistent with that notation just so that we dont have 'info' and 'infos' in the same Variant context.